packages/utils/src/lib/group-by-status.ts

```diff
@@ -0,0 +1,22 @@
+export function groupByStatus<T>(results: PromiseSettledResult<T>[]): {
+  fulfilled: PromiseFulfilledResult<T>[];
+  rejected: PromiseRejectedResult[];
+} {
+  return results.reduce<{
+    fulfilled: PromiseFulfilledResult<T>[];
+    rejected: PromiseRejectedResult[];
+  }>(
+    (accumulator, currentValue) => {
+      if (currentValue.status === 'fulfilled' && 'value' in currentValue) {
+        accumulator.fulfilled.push(currentValue);
+      } else if (
+        currentValue.status === 'rejected' &&
+        'reason' in currentValue
+      ) {
+        accumulator.rejected.push(currentValue);
+      }
+      return accumulator;
+    },
```

Comment:
If using `reduce`, please don't mutate the accumulator.

```suggestion
    (acc, result) => {
      if (result.status === 'fulfilled') {
        return { ...acc, fulfilled: [...acc.fulfilled, result] };
      }
      if (result.status === 'rejected') {
        return { ...acc, rejected: [...acc.rejected, result] };
      }
      return acc;
    },
```